File: bus/internal/bus/topics.go
// SAHA – Event Bus Topic Definitions (Go)
// These constants MUST mirror saha/event_bus/topics.py exactly.
// Switching backends (Redis → Kafka) means changing only the adapter,
// never these topic strings.
package bus

const (
	TopicAgentRequests    = "SAHA/agent_requests"
	TopicProviderResponses = "SAHA/provider_responses"
	TopicEvalInputs       = "SAHA/eval_inputs"
	TopicEvalResults      = "SAHA/eval_results"
	TopicBudgetInterrupts = "SAHA/budget_interrupts"
)

// AllTopics enumerates every known SAHA topic.
var AllTopics = []string{
	TopicAgentRequests,
	TopicProviderResponses,
	TopicEvalInputs,
	TopicEvalResults,
	TopicBudgetInterrupts,
}

File: saha/event_bus/topics.py
"""
SAHA – Event Bus topic definitions.
Topics are stable SAHA contracts; switching from Redis → Kafka
requires only changing the adapter, never the topic names.
"""

# Publisher: Execution Harness  → Subscriber: Vendor Abstraction
AGENT_REQUESTS = "SAHA/agent_requests"

# Publisher: Vendor Abstraction → Subscriber: Execution Harness
PROVIDER_RESPONSES = "SAHA/provider_responses"

# Publisher: Execution Harness  → Subscriber: Eval Harness
EVAL_INPUTS = "SAHA/eval_inputs"

# Publisher: Eval Harness       → Subscriber: Observability / Cost Router
EVAL_RESULTS = "SAHA/eval_results"

# Publisher: Execution Harness  → Subscriber: Vendor Abstraction
BUDGET_INTERRUPTS = "SAHA/budget_interrupts"

# Publisher: Observability (AnomalyDetector) → Subscriber: Routing, HITL, dashboards
ANOMALY_ALERTS = "SAHA/anomaly_alerts"

# All known topics
ALL_TOPICS = [
    AGENT_REQUESTS,
    PROVIDER_RESPONSES,
    EVAL_INPUTS,
    EVAL_RESULTS,
    BUDGET_INTERRUPTS,
    ANOMALY_ALERTS,
]
